x/treasury/keeper/keeper.go

```diff
@@ -387,3 +394,19 @@ func (k Keeper) HasBurnTaxExemptionAddress(ctx sdk.Context, addresses ...string)
 
 	return true
 }
+
+// HasBurnTaxExemptionAddressSmart returns true if a provided address is a
+// smart contract AND is in the tax exemption list
+func (k Keeper) HasBurnTaxExemptionAddrSmart(ctx sdk.Context, address string) bool {
+	contractAddr, err := sdk.AccAddressFromBech32(address)
+	if err != nil {
+		return false
+	}
+
+	info := k.wasmKeeper.GetContractInfo(ctx, contractAddr)
+	if info == nil {
+		return false
+	}
```

Comment:
This logic is already included in the wasm module, does it still need to be checked here?

Reply:
I am sometimes unsure about what I can assume and what not. I will check the GetContractInfo() function once again and remove the check if it would be a double check otherwise.